app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt

```diff
@@ -150,6 +182,52 @@ fun PocketStories(
     }
 }
 
+private fun Modifier.onShown(
+    @FloatRange(from = 0.0, to = 1.0) threshold: Float,
+    onVisible: () -> Unit,
+): Modifier {
+    return composed {
+        val context = LocalContext.current
+        var wasEventReported by remember { mutableStateOf(false) }
+
+        onGloballyPositioned { coordinates ->
+            @Suppress("CollapsibleIfStatements")
+            if (!wasEventReported) {
+                if (coordinates.isVisible(context, threshold)) {
+                    wasEventReported = true
+                    onVisible()
+                }
+            }
+        }
+    }
+}
+
+private fun LayoutCoordinates.isVisible(
+    context: Context,
+    @FloatRange(from = 0.0, to = 1.0) threshold: Float,
+): Boolean {
+    if (!isAttached) return false
```

Comment:
will this work if a story Composable is in a Lazy list and is rendered but just off screen? Curious at what point this function returns `true`, basically.

Reply:
This is exactly the intended scenario - report a composable as shown/visible only if it is currently in the visible bounds of the screen, otherwise if it offscreen this will return false.

Reply:
An interesting aspect is that since not everything on the screen is part of the same (or parent) composable this will not know if another view is shown on top of it (like it happens with the toolbar shown on homescreen). This will unfortunately mean that that data has to be fed in separately.